modules/rotating-bound-service-keys.adoc

```diff
@@ -0,0 +1,404 @@
+----
+
+** For {aws-short} clusters that store the OIDC configuration in a private S3 bucket that is accessed by the IAM identity provider through a public CloudFront distribution URL, complete the following steps:
+
+... Determine the public CloudFront distribution URL by running the following command:
++
+[source,terminal]
+----
+$ basename $(oc get authentication cluster -o jsonpath={'.spec.serviceAccountIssuer'} )
+----
++
+.Example output
+[source,text]
+----
+<subdomain>.cloudfront.net
+----
++
+where `<subdomain>` is an alphanumeric string.
+
+... Determine the private S3 bucket name by running the following command:
++
+[source,terminal]
+----
+$ aws cloudfront list-distributions --query "DistributionList.Items[].{DomainName: DomainName, OriginDomainName: Origins.Items[0].DomainName}[?contains(DomainName, '<subdomain>.cloudfront.net')]"
+----
++
+.Example output
+[source,text]
+----
+[
+    {
+        "DomainName": "<subdomain>.cloudfront.net",
+        "OriginDomainName": "<s3_bucket>.s3.us-east-2.amazonaws.com"
+    }
+]
+----
++
+where `<s3_bucket>` is the private S3 bucket name for your cluster.
+
+... Configure the following environment variables:
++
+[source,text]
+----
+AWS_BUCKET=$<s3_bucket>
+CLUSTER_NAME=$(echo $AWS_BUCKET|awk -F'-oidc' '{print$1}')
+----
++
+where `<s3_bucket>` is the private S3 bucket name for your cluster.
+endif::aws[]
+
+. Create a directory with the name of the `TEMPDIR` environment variable by running the following command:
++
+[source,terminal]
+----
+$ mkdir ${TEMPDIR}
+----
+
+. To cause the Kubernetes API server to create a new bound service account signing key, you delete the next bound service account signing key.
++
+[IMPORTANT]
+====
+After you complete this step, the Kubernetes API server starts to roll out a new key.
+To reduce the risk of authentication failures, complete the remaining steps as quickly as possible.
+The remaining steps might be disruptive to workloads.
+====
++
+When you are ready, delete the next bound service account signing key by running the following command:
++
+[source,terminal]
+----
+$ oc delete secrets/next-bound-service-account-signing-key \
+  -n openshift-kube-apiserver-operator
+----
+
+. Download the public key from the service account signing key secret that the Kubernetes API server created by running the following command:
++
+[source,terminal]
+----
+$ oc get secret/next-bound-service-account-signing-key \
+  -n openshift-kube-apiserver-operator \
+  -ojsonpath='{ .data.service-account\.pub }' | base64 \
+  -d > ${TEMPDIR}/serviceaccount-signer.public
+----
+
+. Use the public key to create a `keys.json` file by running the following command:
+ifdef::aws[]
++
+[source,terminal]
+----
+$ ccoctl aws create-identity-provider \
+  --dry-run \// <1> 
+  --output-dir ${TEMPDIR} \
+  --name fake \// <2>
+  --region us-east-1
+----
+//TODO: Does the region value matter here, or is it a placeholder in dry-run like "fake"?
```

Comment:
Do they need to provide their actual region here?

Reply:
The region is validated, so it needs to be one that actually exists. It does not have to be the one the cluster is in.

For note, I just created an [Epic](https://issues.redhat.com/browse/CCO-639) to provide a ccoctl subcommand that will only create these keys without having to provide all of these arguments It will greatly simplify this part of the process if/when we get to it.